src/Analysis/Engine/Impl/Intellisense/AnalysisQueue.cs

```diff
@@ -55,6 +58,7 @@ private void Enqueue(object key, Func<CancellationToken, Task> handler, Analysis
             RaiseEventOnThreadPool(AnalysisStarted);
             while (!_ppc.IsDisposed) {
                 try {
+                    _queueEnabled.Wait();
```

Comment:
Done. Not sure about the `CancellationToken` since `CountdownDisposable` is not using it. If `ppc` is canceled, it doesn't matter if queue is stopped.